### DISCUSSION TITLE
AssertionError: cost_func should return cost when using SafeLOOP algorithm

### DISCUSSION THREAD

**Discussion by Chip1ne** [PKU-Alignment/omnisafe#347] [Q&A] [answered]
Hi everyone,

I’m trying to train a model using the SafeLOOP algorithm with the omnisafe framework. I’m running the following command in the terminal:

`omnisafe train --algo SafeLOOP --env-id SafetyPointGoal1-v0 --total-steps 1000 --vector-env-nums 1`

However, I encountered the following error:

`AssertionError: cost_func should return cost`

I’m unsure why this error is occurring or how to resolve it. Could anyone provide some guidance on what might be causing this and how I can fix it?

Thanks in advance for your help!

**Comment by hdadong** [comment, marked as answer]
Hello, you should use the 'SafetyCarGoal1-v0-modelbased' for env-id.
Please check the configurations carefully for model-based here: https://github.com/PKU-Alignment/omnisafe/tree/main/benchmarks/model-based

**Reply by Chip1ne** [reply]
Thanks!
